pkg/stdlib/connection_test.go

```diff
@@ -0,0 +1,222 @@
+	require.Error(t, err)
+
+	_, err = db.ExecContext(context.TODO(), "INSERT INTO myTable(id, name) VALUES (2, 'immu2')")
+	require.Error(t, err)
+
+	_, err = db.QueryContext(context.TODO(), "SELECT * FROM myTable")
+	require.Error(t, err)
+}
+
+func TestConn_QueryContext(t *testing.T) {
+	port, cleanup := testServer(t)
+	defer cleanup()
+
+	opts := client.DefaultOptions()
+	opts.WithPort(port)
+	opts.Username = "immudb"
+	opts.Password = "immudb"
+	opts.Database = "defaultdb"
+
+	cli, err := client.NewImmuClient(opts)
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+	_, err = cli.Login(ctx, []byte(opts.Username), []byte(opts.Password))
+	require.NoError(t, err)
+
+	c := Conn{
+		immuClient: cli,
+	}
+
+	table := "mytable"
+	result, err := c.ExecContext(context.TODO(), fmt.Sprintf("CREATE TABLE %s (id INTEGER, amount INTEGER, total INTEGER, title VARCHAR, content BLOB, isPresent BOOLEAN, PRIMARY KEY id)", table), nil)
+	require.NoError(t, err)
+	require.NotNil(t, result)
+
+	binaryContent := []byte("my blob content1")
+	argsV := []driver.NamedValue{
+		{Name: "id", Value: 1},
+		{Name: "amount", Value: 100},
+		{Name: "total", Value: 200},
+		{Name: "title", Value: "title 1"},
+		{Name: "content", Value: binaryContent},
+		{Name: "isPresent", Value: true},
+	}
+	_, err = c.ExecContext(context.TODO(), fmt.Sprintf("INSERT INTO %s (id, amount, total, title, content, isPresent) VALUES (?, ?, ?, ?, ?, ?)", table), argsV)
+	require.NoError(t, err)
+
+	rows, err := c.QueryContext(ctx, "SELECT * FROM myTable limit 1", nil)
+	require.NoError(t, err)
+	defer rows.Close()
+
+	dst := make([]driver.Value, 6, 6)
+	rows.Next(dst)
+
+	require.Equal(t, int64(1), dst[0])
+	require.Equal(t, int64(100), dst[1])
+	require.Equal(t, int64(200), dst[2])
+	require.Equal(t, "title 1", dst[3])
+	require.Equal(t, binaryContent, dst[4])
+	require.Equal(t, true, dst[5])
+}
+
+func TestConn_QueryContextEmptyTable(t *testing.T) {
+	port, cleanup := testServer(t)
+	defer cleanup()
+
+	opts := client.DefaultOptions()
+	opts.WithPort(port)
+	opts.Username = "immudb"
+	opts.Password = "immudb"
+	opts.Database = "defaultdb"
+
+	cli, err := client.NewImmuClient(opts)
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+	_, err = cli.Login(ctx, []byte(opts.Username), []byte(opts.Password))
+	require.NoError(t, err)
+
+	c := Conn{
+		immuClient: cli,
+	}
+
+	table := "emptytable"
+	result, err := c.ExecContext(context.TODO(), fmt.Sprintf("CREATE TABLE %s (id INTEGER, amount INTEGER, total INTEGER, title VARCHAR, content BLOB, isPresent BOOLEAN, PRIMARY KEY id)", table), nil)
+	require.NoError(t, err)
+	require.NotNil(t, result)
+
+	rows, err := c.QueryContext(ctx, "SELECT * FROM emptytable limit 1", nil)
+	require.NoError(t, err)
+	defer rows.Close()
+
+	cols := rows.Columns()
+	require.Equal(t, len(cols), 6)
+}
+
+/*func TestConn_Ping(t *testing.T) {
```

Comment:
is this test case not working?

Reply:
no it doesn't, it was commented out before, so did not want to address the issue in this PR